python2/koans/about_new_style_classes.py

```diff
@@ -22,47 +22,47 @@ class NewStyleClass(object):
         pass
 
     def test_new_style_classes_inherit_from_object_base_class(self):
-        self.assertEqual(____, issubclass(self.NewStyleClass, object))
-        self.assertEqual(____, issubclass(self.OldStyleClass, object))
+        self.assertEqual(True, issubclass(self.NewStyleClass, object))
+        self.assertEqual(False, issubclass(self.OldStyleClass, object))
 
     def test_new_style_classes_have_more_attributes(self):
-        self.assertEqual(__, len(dir(self.OldStyleClass)))
-        self.assertEqual(__, self.OldStyleClass.__doc__)
-        self.assertEqual(__, self.OldStyleClass.__module__)
+        self.assertEqual(2, len(dir(self.OldStyleClass)))
+        self.assertEqual("An old style class", self.OldStyleClass.__doc__)
+        self.assertEqual("koans.about_new_style_classes", self.OldStyleClass.__module__)
 
-        self.assertEqual(__, len(dir(self.NewStyleClass)))
+        self.assertEqual(18, len(dir(self.NewStyleClass)))
         # To examine the available attributes, run
         # 'dir(<Class name goes here>)'
         # from a python console
 
     # ------------------------------------------------------------------
 
     def test_old_style_classes_have_type_but_no_class_attribute(self):
-        self.assertEqual(__, type(self.OldStyleClass).__name__)
+        self.assertEqual("classobj", type(self.OldStyleClass).__name__)
 
         try:
             cls = self.OldStyleClass.__class__.__name__
         except Exception as ex:
             pass
 
         # What was that error message from the exception?
-        self.assertMatch(__, ex[0])
+        self.assertMatch("class OldStyleClass has no attribute '__class__'", ex[0])
 
     def test_new_style_classes_have_same_class_as_type(self):
         new_style = self.NewStyleClass()
-        self.assertEqual(__, self.NewStyleClass.__class__.__name__)
+        self.assertEqual("type", self.NewStyleClass.__class__.__name__)
         self.assertEqual(
-            __,
+            True,
             type(self.NewStyleClass) == self.NewStyleClass.__class__)
 
     # ------------------------------------------------------------------
 
     def test_in_old_style_instances_class_is_different_to_type(self):
         old_style = self.OldStyleClass()
-        self.assertEqual(__, old_style.__class__.__name__)
-        self.assertEqual(__, type(old_style).__name__)
+        self.assertEqual("OldStyleClass", old_style.__class__.__name__)
+        self.assertEqual("instance", type(old_style).__name__)
 
     def test_new_style_instances_have_same_class_as_type(self):
         new_style = self.NewStyleClass()
-        self.assertEqual(__, new_style.__class__.__name__)
-        self.assertEqual(__, type(new_style) == new_style.__class__)
+        self.assertEqual("NewStyleClass", new_style.__class__.__name__)
+        self.assertEqual(True, type(new_style) == new_style.__class__)
```

Comment:
Just one comment about this exercise.

In Python 3, there is only the "new style" of classes. As such, the format used to create "old style" of classes creates new style. I only mention this because in Python 3, you typically use the "old style" format to create classes (do not include the `(object)` in the definition) because the new format is more verbose and achieves exactly the same. 

Always adding `(object)` to your class definitions in your Python 2 code makes it immediately portable to Python 3. This is of course if your class is no inheriting from another class. In that case, `(object)` is no necessary, and you would use `(ParentClass)`.

```python
class ParentClass(object):
  "New style of class."
  def __init__(self, v):
    self.v = v

class ChildClass(ParentClass):
  "Inherits from a new style class, and it is a new style class."
  def __init__(self, v):
    self.v = v * 2